modules/mono/mono_gd/support/android_support.cpp

```diff
@@ -636,7 +635,8 @@ GD_PINVOKE_EXPORT int32_t _monodroid_get_dns_servers(void **r_dns_servers_array)
 
 	if (dns_servers_count > 0) {
 		size_t ret_size = sizeof(char *) * (size_t)dns_servers_count;
-		*r_dns_servers_array = malloc(ret_size); // freed by the BCL
+		*r_dns_servers_array = memalloc(ret_size); // freed by the BCL
+		ERR_FAIL_NULL_MSG(r_dns_servers_array, "Out of memory.");
```

Comment:
Like the comment says, this is freed by the Mono BCL which doesn't use Godot's `memfree` (`Memory::free_static`). Isn't that going to cause trouble?
Honestly, I would prefer this wasn't touched. Mono's allocations don't use Godot's allocation macros any way.

Reply:
See #46807 - if `malloc` fails (e.g. OOM), this can lead to a null pointer dereference.

But yes in this file we can likely add `ERR_FAIL_NULL_MSG` on the result of `malloc()` without changing it to Godot's static alloc.